core/drivers/crypto/caam/utils/utils_mem.c

```diff
@@ -71,21 +71,26 @@ static uint32_t read_cacheline_size(void)
 static void *mem_alloc(size_t size, uint8_t type)
 {
 	void *ptr = NULL;
+	size_t cacheline_size = dcache_get_line_size();
+	size_t alloc_size = size;
 
 	MEM_TRACE("alloc %zu bytes of type %" PRIu8, size, type);
 
+	if (ROUNDUP_OVERFLOW(alloc_size, cacheline_size, &alloc_size))
```

Comment:
Is this only needed for `type & MEM_TYPE_ALIGN`?

Reply:
Indeed, this roundup should only be done when the buffer is aligned. I will fix that

Reply:
Fixed in https://github.com/OP-TEE/optee_os/pull/4407/commits/adc0d4a2c99f54121b0a5077459cdeb9e414afef